server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -838,7 +841,7 @@ private abstract static class IndexOptions implements ToXContent {
             this.type = type;
         }
 
-        abstract KnnVectorsFormat getVectorsFormat();
+        abstract KnnVectorsFormat getVectorsFormat(int mergeThreadCount);
```

Comment:
Something about this really bugs me. The merge thread count is a dynamically updatable value. But doing this, is it really dynamic for the workers in the dense vector field mapper?